labelbox/data/serialization/ndjson/converter.py

```diff
@@ -40,8 +42,45 @@ def serialize(
         Returns:
             A generator for accessing the ndjson representation of the data
         """
+        used_annotation_uuids = set()
+        for label in labels:
+            annotation_uuid_lookup = {}
+            # UUIDs are private properties used to enhance UX when defining relationships.
+            # They are created for all annotations, but only utilized for relationships.
+            # To avoid overwriting, UUIDs must be unique across labels.
+            # Non-relationship annotation UUIDs are dropped (server-side generation will occur).
+            # For relationship annotations, new UUIDs are generated and stored in a lookup table.
+            for annotation in label.annotations:
+                if isinstance(annotation, RelationshipAnnotation):
+                    source_uuid = annotation.value.source._uuid
+                    target_uuid = annotation.value.target._uuid
+
+                    if (len(
+                            used_annotation_uuids.intersection(
+                                {source_uuid, target_uuid})) > 0):
+                        new_source_uuid = uuid.uuid4()
+                        new_target_uuid = uuid.uuid4()
+
+                        annotation_uuid_lookup[source_uuid] = new_source_uuid
```

Comment:
perhaps, rename `annotation_uuid_lookup` to `annotation_uuid_to_generated_uuid_lookup` ? Not sure, just trying to give this important variable even more clarity...